students/juyeob/westagram/users/views.py

```diff
@@ -42,6 +42,30 @@ def post(self, request):
             
             return JsonResponse({"message": "SUCCESS"}, status=201)
             
+        except KeyError:
+            return JsonResponse({"message": "KEY_ERROR"}, status=400)
+        except JSONDecodeError:
+            return JsonResponse({"message": "JSON_DECODE_ERROR"}, status=400)
+
+
+class LogInView(View):
+    def post(self, request):
+        
+        try:
+            data = json.loads(request.body)
+
+            email    = data['email']
+            password = data['password']
+
+            if not User.objects.filter(email = email).exists():
+                return JsonResponse({"message": "INVALID_USER"}, status=401)         
+
+            if not User.objects.filter(password = password).exists():    
+                return JsonResponse({"message": "INVALID_USER"}, status=401)
+
+            if User.objects.filter(email = email).exists() and User.objects.filter(password = password).exists():
```

Comment:
해당 코드와 주엽님의 코드의 차이가 무엇인지 한 번 생각해 보세요.
```suggestion
            if User.objects.get(email = email).password != password:    
                return JsonResponse({"message": "INVALID_USER"}, status=401)
```

Reply:
아 이미 데이터베이스에 다 같이 있으니깐 이메일을 이용해서 해당 로우를 가져온 다음에 패스워드가 일치하는 지 확인하는 과정일까요?

Reply:
네 맞습니다~~!!